packages/salesforcedx-vscode-core/src/commands/forceSourceRetrieveManifest.ts

```diff
@@ -41,14 +35,16 @@ export class LibrarySourceRetrieveManifestExecutor extends RetrieveExecutor<
     response: ContinueResponse<string>
   ): Promise<ComponentSet> {
     const packageDirs = await SfdxPackageDirectories.getPackageDirectoryPaths();
+    const rootWorkspacePath = workspaceUtils.getRootWorkspacePath();
+    const resolveSourcePaths = packageDirs.map(packageDir => join(rootWorkspacePath, packageDir));
 
-    return ComponentSet.fromManifest({
+    const componentSet = await ComponentSet.fromManifest({
       manifestPath: response.data,
-      resolveSourcePaths: packageDirs.map(relativeDir =>
```

Comment:
I'm glad this was moved out! I was confused by the move at first, but I like it better without all the extra logic sneakily nestled in here.

Reply:
Yea, easier to follow, and and much easier to debug, too!